### ISSUE TITLE
k8s always failed to start

### ISSUE THREAD

**Issue by lijian12345** [docker/for-win#13354]
- [ x ] I have tried with the latest version of Docker Desktop
  - [ x ] I have tried disabling enabled experimental features
  - [ x ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior

in 4.17.0, k8s always failed to start. I tried many ways, such as reinstalling it, cleaning data/resetting k8s, modifying .wslconfig to add resources and deleting PKI directory. None of these ways can solve my problem. When I close docker desktop and reopen it, k8s still can't be started.
![Docker_Desktop_Bw3gx8ErsP](https://user-images.githubusercontent.com/26590220/229356044-aa17554b-9aef-4267-b44d-6efc3181c65e.png)
I also tried 4.17.1, the behavior is the same as 4.17.0. The different point is when I close docker desktop and reopen it, even electron is stuck.

### Information

  - Windows Version: win 10 home 19045.2788
  - Docker Desktop Version: 4.17.0
  - WSL2 or Hyper-V backend?  WSL2(Ubuntu 20.04)
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: no

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
```
[2023-04-02T13:36:51.196460000Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```
### Steps to reproduce the behavior

  1. just reinstall docker desktop
  2. and enable k8s

**Comment by BlexToGo** [comment]
I have exactly the same problem. On my machine it started together with https://github.com/microsoft/WSL/issues/9867, but the solution `wsl --update` has not fixed Kubernetes on Docker Desktop 4.17.1 (101757).

The following containers are running when I look at `docker ps`
```
CONTAINER ID   IMAGE                       COMMAND                  CREATED          STATUS          PORTS     NAMES
68c5240f7d05   e3ed7dee73e9                "kube-scheduler --au…"   30 minutes ago   Up 30 minutes             k8s_kube-scheduler_kube-scheduler-docker-desktop_kube-system_1fa929fab6e17047f7779b2ab8125174_49
c58500f113b8   aebe758cef4c                "etcd --advertise-cl…"   30 minutes ago   Up 30 minutes             k8s_etcd_etcd-docker-desktop_kube-system_2449ddc0985e3be8dd23ffc4d12cb53b_46
dcbf180b346d   registry.k8s.io/pause:3.8   "/pause"                 30 minutes ago   Up 30 minutes             k8s_POD_kube-scheduler-docker-desktop_kube-system_1fa929fab6e17047f7779b2ab8125174_46
ad1963f735ff   registry.k8s.io/pause:3.8   "/pause"                 30 minutes ago   Up 30 minutes             k8s_POD_kube-controller-manager-docker-desktop_kube-system_d9c6bbd179b0d64e8d303a659acf3a74_46
23f5a3d11e70   registry.k8s.io/pause:3.8   "/pause"                 30 minutes ago   Up 30 minutes             k8s_POD_kube-apiserver-docker-desktop_kube-system_f76ea91a200a6b1cfe31c7a114460aac_46
9d56bbdeedba   registry.k8s.io/pause:3.8   "/pause"                 30 minutes ago   Up 30 minutes             k8s_POD_etcd-docker-desktop_kube-system_2449ddc0985e3be8dd23ffc4d12cb53b_46
```
But Kubernetes is still not responding when I try to reach it via `kubectl get nodes` for example.

"Reset Kubernetes Cluster" in the Docker Desktop Kubernetes Settings has not helped either.

General WSL and system information (`wsl -v`):
```
WSL-Version: 1.1.6.0
Kernelversion: 5.15.90.1
WSLg-Version: 1.0.50
MSRDC-Version: 1.2.3770
Direct3D-Version: 1.608.2-61064218
DXCore-Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-Version: 10.0.19043.2364
```

**Comment by lijian12345** [comment]
today I tried reinstall wslg(ubuntu 22.04) and docker desktop(4.18.0), then k8s can start successfully. Maybe u can try as this way. And I noticed when k8s started, the memory usage reached about 6G...what a big usage...
